src/test/java/it/pagopa/transactions/services/v1/CircuitBreakerTest.java

```diff
@@ -54,9 +64,315 @@ class CircuitBreakerTest {
     @MockBean
     private TransactionActivateHandler transactionActivateHandlerV1;
 
+    @MockBean
+    private TransactionsViewRepository transactionsViewRepository;
+    @MockBean
+    private TransactionsUtils transactionsUtils;
+
     @Autowired
     private CircuitBreakerRegistry circuitBreakerRegistry;
 
+    @Autowired
+    private RetryRegistry retryRegistry;
+
+    private static final JsonNode resilience4jConfiguration;
+
+    private static final Map<String, Exception> exceptionMapper = Map.of(
+            "it.pagopa.transactions.exceptions.UnsatisfiablePspRequestException",
+            new UnsatisfiablePspRequestException(
+                    new PaymentToken(""),
+                    RequestAuthorizationRequestDto.LanguageEnum.IT,
+                    0
+            ),
+            "it.pagopa.transactions.exceptions.PaymentNoticeAllCCPMismatchException",
+            new PaymentNoticeAllCCPMismatchException("rptId", true, true),
+            "it.pagopa.transactions.exceptions.TransactionNotFoundException",
+            new TransactionNotFoundException(""),
+            "it.pagopa.transactions.exceptions.AlreadyProcessedException",
+            new AlreadyProcessedException(new TransactionId(TransactionTestUtils.TRANSACTION_ID)),
+            "it.pagopa.transactions.exceptions.NotImplementedException",
+            new NotImplementedException(""),
+            "it.pagopa.transactions.exceptions.InvalidRequestException",
+            new InvalidRequestException(""),
+            "it.pagopa.transactions.exceptions.TransactionAmountMismatchException",
+            new TransactionAmountMismatchException(10, 11),
+            "it.pagopa.transactions.exceptions.NodoErrorException",
+            new NodoErrorException(new CtFaultBean()),
+            "it.pagopa.transactions.exceptions.InvalidNodoResponseException",
+            new InvalidNodoResponseException("")
+    );
+
+    static {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        try {
+            resilience4jConfiguration = mapper.readTree(new File("./src/main/resources/application.yml"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionsForRetry(String retryInstanceName) {
+        return StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(
+                        resilience4jConfiguration
+                                .get("resilience4j.retry")
+                                .get("instances")
+                                .get(retryInstanceName)
+                                .get("ignoreExceptions")
+                                .elements(),
+                        Spliterator.ORDERED
+                ),
+                false
+        )
+                .map(json -> {
```

Comment:
[nit] what abount naming this `element`, `exception`,`ignoredException` or something like that?